atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/debug/WritesDigestEmitter.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.debug;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Multimap;
+import com.google.common.collect.Sets;
+import com.palantir.atlasdb.keyvalue.api.Cell;
+import com.palantir.atlasdb.keyvalue.api.KeyValueService;
+import com.palantir.atlasdb.keyvalue.api.TableReference;
+import com.palantir.atlasdb.keyvalue.api.Value;
+import com.palantir.atlasdb.transaction.api.TransactionManager;
+import com.palantir.atlasdb.transaction.service.TransactionService;
+import com.palantir.common.persist.Persistable;
+import com.palantir.common.streams.KeyedStream;
+
+public class WritesDigestEmitter {
+
+    private final TransactionService transactionService;
+    private final TableReference tableReference;
+    private final KeyValueService keyValueService;
+
+    public WritesDigestEmitter(TransactionManager transactionManager, TableReference tableReference) {
```

Comment:
How do we feel about a three argument package-private or private constructor and a static factory method? It seems like passing in the whole `TransactionManager` is more than we actually need.

Reply:
The thing that constructs this (see #4388) has already 5 params (including the transaction manager), don't particularly want to make that thing 6, and if so, then changing the constructor/factory in `WritesDigestEmitter` is not particularly beneficial.